Documentation/operations/upgrade.rst

```diff
@@ -178,56 +175,18 @@ the initial deployment:
 
       helm upgrade cilium |CHART_RELEASE| \\
         --namespace=kube-system \\
-        --set config.upgradeCompatibility=1.7 \\
-        --set agent.keepDeprecatedProbes=true
+        --set keepDeprecatedProbes=true
 
 .. note::
 
    Make sure that you are using the same options as for the initial deployment.
-   Instead of using ``--set``, you can also modify the ``values.yaml`` in
-   ``install/kubernetes/cilium/values.yaml`` and use it to regenerate the YAML
-   for the latest version. Running any of the previous commands will overwrite
-   the existing cluster's `ConfigMap` which might not be ideal if you want to
-   keep your existing `ConfigMap` (see next option).
-
-Step 2: Option B: Preserve ConfigMap
-------------------------------------
-
-Alternatively, you can use `Helm` to regenerate all Kubernetes resources except
-for the `ConfigMap`. The configuration of Cilium is stored in a `ConfigMap`
-called ``cilium-config``. The format is compatible between minor releases so
-configuration parameters are automatically preserved across upgrades. However,
-new minor releases may introduce new functionality that require opt-in via the
-`ConfigMap`. Refer to the :ref:`upgrade_version_specifics` for a list of new
-configuration options for each minor version.
 
-.. include:: ../gettingstarted/k8s-install-download-release.rst
-
-.. tabs::
-  .. group-tab:: kubectl
-
-    Generate the required YAML file and deploy it:
-
-    .. parsed-literal::
-
-      helm template |CHART_RELEASE| \\
-        --namespace kube-system \\
-        --set config.enabled=false \\
-        > cilium.yaml
-      kubectl apply -f cilium.yaml
-
-  .. group-tab:: Helm
-
-    Keeping an existing `ConfigMap` with ``helm upgrade`` is currently not
-    supported.
-
-.. note::
-
-   The above variant can not be used in combination with ``--set`` or providing
-   ``values.yaml`` because all options are fed into the DaemonSets and
-   Deployments using the `ConfigMap` which is not generated if
-   ``config.enabled=false`` or ``config.keepCurrent=true`` are set. The above
-   command *only* generates the DaemonSet, Deployment and RBAC definitions.
+   Instead of using ``--set``, you can also save the values relative to your
+   deployment in a YAML file and use it to regenerate the YAML for the latest
+   Cilium version. Running any of the previous commands will overwrite
+   the existing cluster's `ConfigMap` so it is critical to preserve any existing
+   options, either by setting them at the command line or storing them in a
+   YAML file.
```

Comment:
Should we provide an example of how to correctly upgrade while passing your own values file to ensure the values file overwrites the correct options?

Reply:
done thanks